ocaml/xapi/xapi_pool.ml

```diff
@@ -3698,3 +3698,45 @@ let set_local_auth_max_threads ~__context:_ ~self:_ ~value =
 
 let set_ext_auth_max_threads ~__context:_ ~self:_ ~value =
   Xapi_session.set_ext_auth_max_threads value
+
+let get_auth_filename tmp_dir tarpath =
+  let filename =
+    if String.contains tarpath '/' then
+      Filename.basename tarpath
+    else
+      tarpath
+  in
+  let tmp_auth_file = Filename.concat tmp_dir filename in
+  Helpers.touch_file tmp_auth_file ; tmp_auth_file
+
+let get_guest_secureboot_readiness ~__context ~self =
```

Comment:
This method is a bit complex as it needs to extract the tarball in a pool `(custom)_uefi_certificates` db field in a temp directory and then list them before removing the extracted files...

Another (simpler) solution would be to list auth files in `varstore_dir` as it's supposed to always be in sync with the XAPI DB field unless a user would have manually modified it which wouldn't be supported.
What do you guys think?

Reply:
As long as its limitations are documented, I don't mind using the method that doesn't rely on creating files